src/daft-micropartition/src/ops/cast_to_schema.rs

```diff
@@ -20,9 +20,9 @@ impl MicroPartition {
         let guard = self.state.lock().unwrap();
         match guard.deref() {
             // Replace schema if Unloaded, which should be applied when data is lazily loaded
-            TableState::Unloaded(scan_task_batch) => Ok(MicroPartition::new_unloaded(
+            TableState::Unloaded(scan_task) => Ok(MicroPartition::new_unloaded(
                 schema.clone(),
-                scan_task_batch.clone(),
+                scan_task.clone(),
```

Comment:
this looks like an expensive clone. (We should also probably remove Clone from ScanTask). We should make TabularScan hold an Arc<ScanTask> instead unless we need the mutability